Mathlib/Data/Finset/NatAntidiagonal.lean

```diff
@@ -122,10 +128,35 @@ theorem filter_snd_eq_antidiagonal (n m : ℕ) :
     filter (fun x : ℕ × ℕ ↦ x.snd = m) (antidiagonal n) = if m ≤ n then {(n - m, m)} else ∅ := by
   have : (fun x : ℕ × ℕ ↦ (x.snd = m)) ∘ Prod.swap = fun x : ℕ × ℕ ↦ x.fst = m := by
     ext; simp
-  rw [← map_swap_antidiagonal]
-  simp [filter_map, this, filter_fst_eq_antidiagonal, apply_ite (Finset.map _)]
+  rw [← map_swap_antidiagonal, filter_map]
+  simp [this, filter_fst_eq_antidiagonal, apply_ite (Finset.map _)]
 #align finset.nat.filter_snd_eq_antidiagonal Finset.Nat.filter_snd_eq_antidiagonal
 
+@[simp] lemma antidiagonal_filter_le_fst_of_le {n k : ℕ} (h : k ≤ n) :
+    (antidiagonal n).filter (fun a ↦ k ≤ a.fst) = (antidiagonal (n - k)).map
```

Comment:
There's another version with `(antidiagonal n).filter (fun a ↦ a.fst ≤ k) = (antidiagonal k)).map` that's also true, right? Is it of any use to you?

Reply:
Thanks!